inference/core/workflows/core_steps/sinks/roboflow/model_monitoring_inference_aggregator/v1.py

```diff
@@ -0,0 +1,317 @@
+            self._background_tasks.add_task(registration_task)
+        elif fire_and_forget and self._thread_pool_executor:
+            self._thread_pool_executor.submit(registration_task)
+        else:
+            error_status, message = registration_task()
+        self._cache.set(self._last_report_time_cache_key, datetime.now().isoformat())
+        return {
+            "error_status": error_status,
+            "message": message,
+        }
+
+    def _is_in_reporting_range(self, frequency: int) -> bool:
+        now = datetime.now()
+        last_report_time_str = self._cache.get(self._last_report_time_cache_key)
+        if last_report_time_str is None:
+            self._cache.set(self._last_report_time_cache_key, now.isoformat())
+            last_report_time = now
+        else:
+            last_report_time = datetime.fromisoformat(last_report_time_str)
+        time_elapsed = int((now - last_report_time).total_seconds())
+        return time_elapsed >= int(frequency)
+
+
+# TODO: maybe make this a helper or decorator, it's used in multiple places
+def get_workspace_name(
+    api_key: str,
+    cache: BaseCache,
+) -> str:
+    api_key_hash = hashlib.md5(api_key.encode("utf-8")).hexdigest()
+    cache_key = f"workflows:api_key_to_workspace:{api_key_hash}"
+    cached_workspace_name = cache.get(cache_key)
+    if cached_workspace_name:
+        return cached_workspace_name
+    workspace_name_from_api = get_roboflow_workspace(api_key=api_key)
+    cache.set(key=cache_key, value=workspace_name_from_api, expire=900)
+    return workspace_name_from_api
+
+
+def send_to_model_monitoring_request(
+    cache: BaseCache,
+    last_report_time_cache_key: str,
+    api_key: str,
+    predictions: List[Prediction],
+) -> Tuple[bool, str]:
+    workspace_id = get_workspace_name(api_key=api_key, cache=cache)
+    try:
+        inference_data = {
+            "timestamp": datetime.now().isoformat(),
+            "source": "workflow",
+            "source_info": "ModelMonitoringInferenceAggregatorBlockV1",
+            "inference_results": [],
+        }
+        if DEVICE_ID:
+            inference_data["device_id"] = DEVICE_ID
+        system_info = get_system_info()
+        if system_info:
+            for key, value in system_info.items():
+                inference_data[key] = value
+        inference_data["inference_results"] = predictions
+        send_inference_results_to_model_monitoring(
+            api_key, workspace_id, inference_data
+        )
+        cache.set(last_report_time_cache_key, datetime.now().isoformat())
+        return (
+            False,
+            "Data sent successfully",
+        )
+    except Exception as error:
+        logging.warning(f"Could not upload inference data. Reason: {error}")
+        return (
+            True,
+            f"Error while uploading inference data. Error type: {type(error)}. Details: {error}",
+        )
+
+
+def format_sv_detections_for_model_monitoring(
```

Comment:
please change the name to reflect that `sv.Detections` and cls results are handled